discovery/src/main/java/com/ticketmaster/api/discovery/DiscoveryApiConfiguration.java

```diff
@@ -0,0 +1,77 @@
+package com.ticketmaster.api.discovery;
+
+import static com.ticketmaster.api.discovery.util.Preconditions.checkArgument;
+import static com.ticketmaster.api.discovery.util.Preconditions.checkIllegalCharacters;
+import static com.ticketmaster.api.discovery.util.Preconditions.checkNotEmpty;
+import static com.ticketmaster.api.discovery.util.Preconditions.checkNotNull;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.regex.Pattern;
+
+import com.ticketmaster.api.discovery.util.Preconditions;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+
+@Getter
+@ToString
+@EqualsAndHashCode
+public class DiscoveryApiConfiguration {
+
+  private String protocol = "https";
+  private String domainName = "app.ticketmaster.com";
+  private Integer port = null;
+  private String apiPackage = "discovery";
+  private String apiVersion = "v2";
+  private String defaultLocale = null;
+
+  public static DiscoveryApiConfigurationBuilder builder() {
+    return new DiscoveryApiConfiguration().new DiscoveryApiConfigurationBuilder();
+  }
+
+  private DiscoveryApiConfiguration() {}
+
+  public int getPort() {
+    return port.intValue();
+  }
+
+  public boolean isPortSet() {
+    return port != null;
+  }
+
+  public class DiscoveryApiConfigurationBuilder {
+
+    public DiscoveryApiConfigurationBuilder proxyHost(String host) {
```

Comment:
I think we should remove the `proxy` term from these configuration as it doesn't act on the `OkHttpClient.proxy`
Could you rename all that for host, port, protocol?